qadence_libs/qinfo_tools/qfi.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from typing import Iterable
+
+import torch
+from qadence import Overlap
+from qadence.blocks import parameters, primitive_blocks
+from qadence.circuit import QuantumCircuit
+from qadence.types import BackendName, DiffMode, OverlapMethod
+from torch import Tensor
+
+from qadence_libs.qinfo_tools.spsa import spsa_2gradient_step
+from qadence_libs.qinfo_tools.utils import hessian
+
+
+def _symsqrt(A: Tensor) -> Tensor:
+    """Computes the square root of a Symmetric or Hermitian positive definite matrix.
+
+    or batch of matrices.
+
+    Code from https://github.com/pytorch/pytorch/issues/25481#issuecomment-1032789228
+    """
+    L, Q = torch.linalg.eigh(A)
+    zero = torch.zeros((), device=L.device, dtype=L.dtype)
+    threshold = L.max(-1).values * L.size(-1) * torch.finfo(L.dtype).eps
+    L = L.where(L > threshold.unsqueeze(-1), zero)  # zero out small components
+    return (Q * L.sqrt().unsqueeze(-2)) @ Q.mH
+
+
+def _set_circuit_vparams(circuit: QuantumCircuit, vparams_values: Iterable) -> None:
+    """Sets the variational parameter values of the circuit."""
+    if vparams_values is not None:
+        blocks = primitive_blocks(circuit.block)
+        iter_index = iter(range(len(blocks)))
+        for block in blocks:
+            params = parameters(block)
+            for p in params:
+                if p.trainable:
+                    p.value = float(vparams_values[next(iter_index)])  # type: ignore
+
+
+def _get_fm_dict(circuit: QuantumCircuit) -> dict:
```

Comment:
@dominikandreasseitz We have something built-in Qadence for that I think ?

Reply:
actually not directly

Reply:
There is the reset_vparams method in the QuantumModel

Reply:
@dominikandreasseitz what do you think?